src/library_wasmfs_fetch.js

```diff
@@ -38,31 +34,71 @@ addToLibrary({
         } catch (e) {
         }
       }
-      var response = await fetch(url);
+      var chunkSize = __wasmfs_fetch_get_chunk_size(file);
+      offset = offset || 0;
+      len = len || chunkSize;
+      var firstChunk = (offset / chunkSize) | 0;
+      var lastChunk = ((offset+len) / chunkSize) | 0;
+      if (!(file in wasmFS$JSMemoryRanges)) {
+        var fileInfo = await fetch(url,{method:"HEAD", headers:{"Range": "bytes=0-"}});
+        if(fileInfo.ok &&
+           fileInfo.headers.has("Content-Length") &&
+           fileInfo.headers.get("Accept-Ranges") == "bytes" &&
+           (parseInt(fileInfo.headers.get("Content-Length")) > chunkSize*2)) {
+          wasmFS$JSMemoryRanges[file] = {size:parseInt(fileInfo.headers.get("Content-Length")), chunks:[], chunkSize:chunkSize};
+        } else {
+          // may as well/forced to download the whole file
+          var wholeFileReq = await fetch(url);
+          if(!wholeFileReq.ok) {
+            throw wholeFileReq;
+          }
+          var wholeFileData = new Uint8Array(await wholeFileReq.arrayBuffer());
+          var text = new TextDecoder().decode(wholeFileData);
+          wasmFS$JSMemoryRanges[file] = {size:wholeFileData.byteLength, chunks:[wholeFileData], chunkSize:wholeFileData.byteLength};
+          return Promise.resolve();
+        }
+      }
+      var allPresent = true;
+      var i;
+      if(lastChunk * chunkSize < offset+len) {
```

Comment:
Space after `if` and `for` keywords here an elsewhere.

Spaces around `+` operator.

Reply:
I think I got them all.